File: Python/Client_U.py
import grpc
import Hello_pb2
import Hello_pb2_grpc

host = grpc.insecure_channel("localhost:50051")

client = Hello_pb2_grpc.TestStub(host)

request = Hello_pb2.RequestType()  ## 把想發出去的 request參數 封裝
request.name = "傑尼龜"
request.content = "J ni J ni"

res = client.SayHello0(request).sentence

print(res)

File: Go/Hello/Client_U/Client_U.go
package main

import (
	"Hello/Hello/Hello"
	"context"
	"fmt"
	"google.golang.org/grpc"
	"log"
	"time"
)

func ClientU(client Hello.TestClient, name string, content string) (response *Hello.ResponseType) {

	request := &Hello.RequestType{Name: name, Content: content} // 封裝成 protobuf 格式

	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second) // 十秒後 啟動withdeadline 結束該function

	defer cancel() // 函數結束 斷掉 context

	response, err := client.SayHello0(ctx, request) // 啟動server 的方法

	if err != nil {
		log.Fatalf("%v.get SayHello0(_) = _ , %v: ", client, err)
	}

	return response

}

func main() {

	serverhost := "localhost:50051"
	conn, err := grpc.Dial(serverhost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("fail to dial: %v", err)
	}
	defer conn.Close()
	client := Hello.NewTestClient(conn)

	// 連線設定




	res := ClientU(client, "傑尼龜", "J ni J ni")

	fmt.Printf("%v",res.Sentence)

}
